Dockerfile

```diff
@@ -45,6 +45,7 @@ RUN \
       ###libomp-dev \
     # Other "our" apt installs (development and testing)
       build-essential \
+      clang-15 \
```

Comment:
In ci.yaml matrix we use clang-runtime==17. Can you use the newer version (17) of the clang runtime here as well? It should be able to since the CI test passes with version 17.

Reply:
I used clang-15 as that is the latest version of clang I could find in the apt repository which is available for Ubuntu 22.04 .  Also when building llvm-project I had to build version 16 as building version 17 and using this with CppInterOp would build but fail with the command on an arm architechture (I cannot remember off the top of my head what message it failed with).  It is not currently possible to switch this build of llvm-project to version 17, as although it passes on x86, it doesn't pass on arm (from tests on my local machine).
https://github.com/compiler-research/xeus-clang-repl/blob/ce09338007b7096c52830b2c8da5671707d08aec/Dockerfile#L329

Reply:
Oh, yes. This is a good one. I am misled that conda-forge also has the 17th version, but this is APT.

Reply:
I could put a fixme next the part of the dockerfile which builds llvm-project version 16.x , until the issue using 17.x on arm is resolved (I'd raise it as an issue, so that there is a log of what the error is), and put a fixme next to using Clang-15 from apt (when the next LTS version of Ubuntu comes out April next year this would have the latest version of Clang since its in apt repository for version 23.10 of Ubuntu, so would be a quick fix).

Reply:
:+1: